syncapi/storage/tables/relations_test.go

```diff
@@ -184,3 +199,53 @@ func TestRelationsTable(t *testing.T) {
 		}
 	})
 }
+
+const threadRelType = "m.thread"
+
+func TestThreads(t *testing.T) {
+	var err error
+	ctx := context.Background()
+	alice := test.NewUser(t)
+	room := test.NewRoom(t, alice)
+
+	firstEvent := room.CreateAndInsert(t, alice, "m.room.message", map[string]interface{}{
+		"body": "first message",
+	})
+	threadReplyEvent := room.CreateAndInsert(t, alice, "m.room.message", map[string]interface{}{
+		"body": "thread reply",
+		"m.relates_to": map[string]interface{}{
+			"event_id": firstEvent.EventID(),
+			"rel_type": threadRelType,
+		},
+	})
```

Comment:
This is an in-memory room, so when you query the relation below, there won't be any events in the database. This means you'll also need to create a `newOutputRoomEventsTable(t, dbType)` below, like you did with the relations table. Then insert the events.